sasmodels/convert.py

```diff
@@ -280,34 +294,42 @@ def _hand_convert(name, oldpars):
 
     return oldpars
 
-def convert_model(name, pars, use_underscore=False):
+def convert_model(name, pars, use_underscore=False, model_version=(3,1,2)):
     """
     Convert model from old style parameter names to new style.
     """
-    newname = _conversion_target(name)
-    if newname is None:
-        return name, pars
-    if ':' in newname:   # core_shell_ellipsoid:1
-        model_info = load_model_info(newname[:-2])
-        # Know that the table exists and isn't multiplicity so grab it directly
-        # Can't use _get_translation_table since that will return the 'bare'
-        # version.
-        translation = CONVERSION_TABLE[newname][1]
-    else:
-        model_info = load_model_info(newname)
-        translation = _get_translation_table(model_info)
-    newpars = _hand_convert(newname, pars.copy())
-    newpars = _convert_name(translation, newpars)
-    newpars = _convert_pars(newpars, translation)
-    if not model_info.structure_factor:
-        newpars = _rescale_sld(model_info, newpars, 1e6)
-    newpars.setdefault('scale', 1.0)
-    newpars.setdefault('background', 0.0)
-    if use_underscore:
-        newpars = _pd_to_underscores(newpars)
+    newpars = pars
+    keys = sorted(CONVERSION_TABLE.keys())
+    for i, version in enumerate(keys):
+        # If the save state is from a later version, skip the check
+        if model_version > version and model_version != keys[i+1]:
+            continue
+        newname = _conversion_target(name, version)
+        # If no conversion is found, move on
+        if newname is None:
+            newname = name
+            continue
+        if ':' in newname:   # core_shell_ellipsoid:1
+            model_info = load_model_info(newname[:-2])
+            # Know the table exists and isn't multiplicity so grab it directly
+            # Can't use _get_translation_table since that will return the 'bare'
+            # version.
+            translation = CONVERSION_TABLE.get(version)[newname][1]
+        else:
+            model_info = load_model_info(newname)
+            translation = _get_translation_table(model_info, version)
+        newpars = _hand_convert(newname, newpars, version)
+        newpars = _convert_pars(newpars, translation)
+        # TODO: Still not convinced this is the best check
+        if not model_info.structure_factor and version == (3,1,2):
+            newpars = _rescale_sld(model_info, newpars, 1e6)
+        newpars.setdefault('scale', 1.0)
+        newpars.setdefault('background', 0.0)
+        if use_underscore:
+            newpars = _pd_to_underscores(newpars)
+        name = newname
     return newname, newpars
 
-
 # ========= BACKWARD CONVERSION sasmodels => sasview 3.x ===========
```

Comment:
I think the code below here is broken, and could be removed.    Because CONVERSION_TABLE is now a two-level table, `revert_name` will return None, and all the tests will claim to pass because none of them are being run.